### PR TITLE
Updated Backend

### PR DESCRIPTION
**Pull request by fuzzyers**
Changes
- get-data route
  - Will get all data from the last month of a chosen data type.
  - add a query parameter `?type={givenDataType}` to the route to choose what data you want from the last month
    - Our types currently `'temperature', 'pressure', 'humidity', 'wind', 'rain_gauge', 'dust', 'co2', 'and gas'` 
  - Rate is Limiting added to this with 10 requests in a 15-minute time span
  - There is also authentication (Bearer)
    - Add to the env file `PASSWORD="PASSWORD"`
- {get-datas}
  - The old get data that are returning single values will now be rate limited. 
- recieve data 
  - The recieve data route has also received authentication
    - Add to the env file `PASSWORD="PASSWORD"`
  - Changed the code to make the file smaller

### CODE REVIEW

**Comment by SamanthaCra** [comment]
image
image
Working as intended, tested temperature and humidity routes.

**Comment by SamanthaCra** [comment]
Could you please clarify if the max requests of 50 is over all of the routes? I did not do 50 retries. 
Also, what is the timer to reset this?

File: fullstack/src/app/api/dust-data/route.js
Comment (SamanthaCra) [inline fullstack/src/app/api/dust-data/route.js:1]: See comments left on overall pull request.

File: fullstack/src/app/api/get-data/route.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        const allowedTypes = ['temperature', 'pressure', 'humidity', 'wind', 'rain_gauge', 'dust', 'co2', 'gas'];
+        
+        if (!allowedTypes.includes(type)) {
+            return new Response(JSON.stringify({ error: 'Invalid data type requested' }), {
+                status: 400,
+                headers: { 'Content-Type': 'application/json' },
+            });
+        }
+        
+        const query = `SELECT * FROM ${type} WHERE timestamp >= NOW() - INTERVAL '30 days';`
```
Comment (SamanthaCra) [inline fullstack/src/app/api/get-data/route.js:41]: Let me know if you want a database function made for this.

File: fullstack/src/app/api/recieve-data/route.js
Comment (SamanthaCra) [inline fullstack/src/app/api/recieve-data/route.js:1]: This is good, only got to test dust, temperature and gas before I got timed out.

**Comment by SamanthaCra** [comment]
Env password is not working.

**Comment by fuzzyers** [comment]
> Could you please clarify if the max requests of 50 is over all of the routes? I did not do 50 retries. Also, what is the timer to reset this?

it is each route and the timer is on 15 minute intervals.

File: fullstack/src/app/utils/ratelimit.js
Comment (SamanthaCra) [inline fullstack/src/app/utils/ratelimit.js:1]: Works as intended. Was able to access data after 15 minutes.

**Review by TomMc608 — APPROVED** [review]
Nice looks good! Thanks

**Pull request merged**